azurelinuxagent/common/protocol/restapi.py

```diff
@@ -177,17 +177,6 @@ def dependency_level_sort_key(self):
             level = (0 - level) - 1
         return level
 
-    def __eq__(self, other):
```

Comment:
This was moved to ExtensionsGoalState.compare, which allows me to add extra debug information to make it easier to follow up any differences in the goal states.